river/main.zig

```diff
@@ -140,9 +140,17 @@ fn defaultInitPath() !?[:0]const u8 {
     };
 
     os.accessZ(path, os.X_OK) catch |err| {
-        std.log.err("failed to run init executable {s}: {s}", .{ path, @errorName(err) });
-        util.gpa.free(path);
-        return null;
+        switch (err) {
+            error.PermissionDenied => {
+                std.log.err("failed to run init executable {s}: the file is not executable", .{path});
```

Comment:
Hmm, I don't think this message is always accurate. EPERM might be returned because one of the directories in the path isn't readable by the user.

Reply:
If we test for `R_OK` first and then for `X_OK`, we could handle that case as well.